### ISSUE TITLE
enumerate(self) type unknown

### ISSUE THREAD

**Issue by fried** [microsoft/pylance-release#423]
## Environment data

-   Language Server version: 2020.9.6
-   OS and version: macOS 10.15.6
-   Python version (& distribution if applicable, e.g. Anaconda): cpython3.8.5

## Expected behaviour

for i, value in enumerate(self):  # (variable) value: bool
   ...

## Actual behaviour

(variable) value: Unknown
Type of "value" is unknownPylance (reportUnknownVariableType)

But the following works just fine

```python
for i, value in enumerate(BitBlob()):   # (variable) value: bool
     pass 
```

## Code Snippet / Additional information

```python
class BitBlob:
         def bad_types(self) -> None:
                for i, value in enumerate(self):
                      pass

         def good_types(self) -> None:
               for i, value in enumerate(BitBlob()):
                      pass

         def __iter__(self) -> Iterator[bool]:  ...


```

**Comment by fried** [comment]
I have a workaround

```python

def good_types(self: BitBlob) -> None:
    for i, value in enumerate(self):
         pass
```

**Comment by jakebailey** [comment]
I can reproduce this.

![image](https://user-images.githubusercontent.com/5341706/94319700-198ecc00-ff40-11ea-859b-a1ecbbfd5313.png)

![image](https://user-images.githubusercontent.com/5341706/94319707-1dbae980-ff40-11ea-9b3e-eac3d63b126e.png)

@erictraut

**Comment by erictraut** [comment]
Thanks for reporting the problem. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.9.7, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202097-30-september-2020
